ui/src/app/subnets/views/SubnetDetails/SubnetDetailsHeader/SubnetActionForms/components/DeleteSubnet/DeleteSubnet.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  Button,
+  Strip,
+  Notification,
+  Row,
+  Col,
+  ActionButton,
+} from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+import { useHistory } from "react-router";
+
+import VisuallyHidden from "app/base/components/VisuallyHidden";
+import { useCycled } from "app/base/hooks";
+import { useId } from "app/base/hooks/base";
+import { actions as subnetActions } from "app/store/subnet";
+import { default as subnetSelectors } from "app/store/subnet/selectors";
+import { getCanBeDeleted, getIsDHCPEnabled } from "app/store/subnet/utils";
+import { default as vlanSelectors } from "app/store/vlan/selectors";
+import urls from "app/subnets/urls";
+import type { SubnetActionProps } from "app/subnets/views/SubnetDetails/types";
+import { formatErrors } from "app/utils";
+
+export const DeleteSubnet = ({
+  setActiveForm,
+}: Omit<SubnetActionProps, "activeForm">): JSX.Element | null => {
+  const subnet = useSelector(subnetSelectors.active);
+  const vlans = useSelector(vlanSelectors.all);
```

Comment:
It might be best to dispatch `vlanActions.fetch()` from this component so we know the VLANs have been loaded.